website/blog/2023-03-10-databend-weekly-84.md

```diff
@@ -0,0 +1,178 @@
+
+```sql
+#> CREATE TABLE employees(id INT, name VARCHAR, salary INT);
+#> REPLACE INTO employees (id, name, salary) ON (id) VALUES (1, 'John Doe', 50000);
+#> SELECT  * FROM Employees;
++------+----------+--------+
+| id   | name     | salary |
++------+----------+--------+
+|    1 | John Doe |  50000 |
++------+----------+--------+
+```
+
+If you want to learn more details about `REPLACE INTO` statement, please read the materials listed below.
+
+- [Docs | DML Commands - REPLACE](https://databend.rs/doc/sql-commands/dml/dml-replace)
+- [PR | feat: replace into statement](https://github.com/datafuselabs/databend/pull/10191)
+
+Also, we plan to gradually transition to the `MERGE INTO` statement based on this, and if you are interested in this topic, you can also follow this issue:
+
+- [Issue | tracking: `MERGE` #10373](https://github.com/datafuselabs/databend/issues/10373)
+
+### RFC: Transform During Load
+
+Currently, we support stage table function, so we can transform when insert from stage into table.
+
+```sql
+insert into table1 from (select c1， c2 from @stage1/path/to/dir);
+```
+
+but this still missing one feature of `COPY INTO`: incremental load (remember files already loaded).
+
+We plan to support transform when copying from stage to table, which enables a series of features such as incremental load and auto cast.
+
+If you are interested in this topic, you can read the following two RFCs:
+
+- [Docs | RFCs - Transform During Load](https://databend.rs/doc/contributing/rfcs/transform-during-load)
+- [Docs | RFCs - Stage With Schema](https://databend.rs/doc/contributing/rfcs/stage-with-schema)
+
+## Code Corner
+
+Discover some fascinating code snippets or projects that showcase our work or learning journey.
+
+### Rust Toolchain: 🦀 v1.70.0-nightly (03-10)
+
+Databend has upgraded its Rust toolchain to `nightly-2023-03-10`. Please remember to run `cargo clean` after pulling the latest code.
+
+We fixed some clippy warnings that we encountered. One notable thing is that we now use `#[default]` to annotate the default value of an `enum`, as shown below.
+
+```rust
+#[derive(Debug, Default)]
+pub enum RecordDelimiter {
+    #[default]
+    Crlf,
+    Any(u8),
+}
+```
+
+If you are interested in this Rust trick, you can read this RFC: [`derive_default_enum`](https://rust-lang.github.io/rfcs/3107-derive-default-enum.html).
+
+### Announcing OpenDAL v0.30
+
+[OpenDAL](https://github.com/datafuselabs/opendal) brings some important changes in its newly released 0.30 version, including removing `Object` Abstraction and bindings for JavaScript and Python.
+
+**Removing `Object` Abstraction**
+
+The term `Object` is applied in various fields, making it difficult to provide a concise definition of `opendal::Object`. We eliminated the intermediate API layer of `Object` and enabled users to directly utilize `Operator`. For example:
```

Comment:
The term `Object` was applied in various fields, so it used to be difficult to provide a concise definition of `opendal::Object`.

Reply:
We eliminated the intermediate API layer of `Object` so users can now directly utilize the `Operator`, for example: